eos/utils/options.cc

```diff
@@ -240,6 +240,35 @@ namespace eos
         return _value;
     }
 
+    BooleanOption::BooleanOption(const Options & options, const std::vector<OptionSpecification> & specifications, const std::string & key) :
+        SpecifiedOption(options, specifications, key)
+    {
+    }
+
+    BooleanOption::~BooleanOption() = default;
+
+    bool
+    BooleanOption::value() const
+    {
+        static const std::map<std::string, bool> map
+        {
+            { "true",  true  },
+            { "false", false }
+        };
+
+        const auto i = map.find(_value);
+        if (map.cend() == i)
+            throw InternalError("Invalid boolean option '" + _value + "' encountered in BooleanOption::value()");
+
+        return i->second;
+    }
+
+    const std::string &
+    BooleanOption::str() const
+    {
+        return _value;
```

Comment:
I would store the boolean value (see above) and return here: ``stringify(_value)``.